### PR TITLE
Add macro flag for unifex multiple interfaces support

### PR DESCRIPTION
**Pull request by mickel8**
related with membraneframework/unifex#43

### CODE REVIEW

File: lib/bundlex/toolchain/common/unix.ex [hunk 1]
```diff
@@ -36,6 +36,23 @@ defmodule Bundlex.Toolchain.Common.Unix do
...
+  defp resolve_compiler_flags(compiler_flags, native_interface) do
+    compiler_flags
+    |> add_interface_macro_flag(native_interface)
+    |> Enum.join(" ")
+  end
+
+  defp add_interface_macro_flag(compiler_flags, native_interface) do
+    macro_flag =
+      cond do
+        native_interface == nil || native_interface == :nif -> "-DNIF"
```
Comment (mat-hek) [inline lib/bundlex/toolchain/common/unix.ex:48]: why is that needed?

File: lib/bundlex/toolchain/common/unix.ex [hunk 2]
```diff
@@ -36,6 +36,23 @@ defmodule Bundlex.Toolchain.Common.Unix do
...
+    |> add_interface_macro_flag(native_interface)
+    |> Enum.join(" ")
+  end
+
+  defp add_interface_macro_flag(compiler_flags, native_interface) do
+    macro_flag =
+      cond do
+        native_interface == :nif -> "-DNIF"
+        native_interface == :cnode -> "-DCNODE"
+        native_interface == :port -> "-DPORT"
```
Comment (mat-hek) [inline lib/bundlex/toolchain/common/unix.ex:50]: - this will fail when `native_interface` is nil instead of adding no flag
- flags for interfaces can be generated with `"-D#{native_interface |> Atom.to_string() |> String.upcase()}"`

Comment (mat-hek) [reply lib/bundlex/toolchain/common/unix.ex:50]: and prefix the defines with `BUDNLEX_`, like `BUNDLEX_CNODE`

File: lib/bundlex/toolchain/common/unix.ex (hunk 1 above)
Comment (mickel8) [reply lib/bundlex/toolchain/common/unix.ex]: fixed in aa9dbd3

File: lib/bundlex/toolchain/common/unix.ex (hunk 2 above)
Comment (mickel8) [reply lib/bundlex/toolchain/common/unix.ex]: fixed in aa9dbd3
